aws/sra-test/integration-tests/aws-sdk-s3/benches/middleware_vs_orchestrator.rs

```diff
@@ -0,0 +1,285 @@
+        <Size>434234</Size>
+        <StorageClass>STANDARD</StorageClass>
+    </Contents>
+</ListBucketResult>
+"#,
+            )
+            .unwrap()
+    })
+}
+
+fn middleware_bench(c: &mut Criterion) {
+    let conn = test_connection();
+    let config = s3::Config::builder()
+        .credentials_provider(s3::config::Credentials::for_tests())
+        .region(s3::config::Region::new("us-east-1"))
+        .http_connector(conn.clone())
+        .build();
+    let client = s3::Client::from_conf(config);
+    c.bench_function("middleware", move |b| {
+        b.to_async(tokio::runtime::Runtime::new().unwrap())
+            .iter(|| async { middleware(&client).await })
+    });
+}
+
+fn orchestrator_bench(c: &mut Criterion) {
+    let conn = test_connection();
+
+    c.bench_function("orchestrator", move |b| {
+        b.to_async(tokio::runtime::Runtime::new().unwrap())
+            .iter(|| async { orchestrator(&conn).await })
+    });
+}
+
+mod orchestrator {
+    use aws_credential_types::cache::{CredentialsCache, SharedCredentialsCache};
+    use aws_credential_types::provider::SharedCredentialsProvider;
+    use aws_credential_types::Credentials;
+    use aws_http::user_agent::{ApiMetadata, AwsUserAgent};
+    use aws_runtime::recursion_detection::RecursionDetectionInterceptor;
+    use aws_runtime::user_agent::UserAgentInterceptor;
+    use aws_sdk_s3::config::Region;
+    use aws_sdk_s3::operation::list_objects_v2::ListObjectsV2Input;
+    use aws_smithy_client::erase::DynConnector;
+    use aws_smithy_runtime::client::connections::adapter::DynConnectorAdapter;
+    use aws_smithy_runtime_api::client::endpoints::StaticUriEndpointResolver;
+    use aws_smithy_runtime_api::client::interceptors::{
+        Interceptor, InterceptorContext, InterceptorError, Interceptors,
+    };
+    use aws_smithy_runtime_api::client::orchestrator::{
+        BoxError, ConfigBagAccessors, Connection, HttpRequest, HttpResponse, TraceProbe,
+    };
+    use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
+    use aws_smithy_runtime_api::config_bag::ConfigBag;
+    use aws_types::region::SigningRegion;
+    use aws_types::SigningService;
+    use http::Uri;
+    use std::sync::Arc;
+
+    pub struct ManualServiceRuntimePlugin(pub DynConnector);
+
+    impl RuntimePlugin for ManualServiceRuntimePlugin {
+        fn configure(&self, cfg: &mut ConfigBag) -> Result<(), BoxError> {
+            let identity_resolvers =
+                aws_smithy_runtime_api::client::orchestrator::IdentityResolvers::builder()
+                    .identity_resolver(
+                        aws_runtime::auth::sigv4::SCHEME_ID,
+                        aws_runtime::identity::credentials::CredentialsIdentityResolver::new(
+                            SharedCredentialsCache::new(CredentialsCache::lazy().create_cache(
+                                SharedCredentialsProvider::new(Credentials::for_tests()),
+                            )),
```

Comment:
Another optimization can be made here: the credential cache should not be getting recreated for every request.